icu4c/source/common/loclikely.cpp

```diff
@@ -75,62 +68,48 @@ createTagStringWithAlternates(
     int32_t trailingLength,
     icu::ByteSink& sink,
     UErrorCode* err) {
-
     if (U_FAILURE(*err)) {
-        goto error;
+        return;
     }
-    else if (langLength >= ULOC_LANG_CAPACITY ||
-             scriptLength >= ULOC_SCRIPT_CAPACITY ||
-             regionLength >= ULOC_COUNTRY_CAPACITY) {
-        goto error;
+
+    if (langLength >= ULOC_LANG_CAPACITY ||
+        scriptLength >= ULOC_SCRIPT_CAPACITY ||
+        regionLength >= ULOC_COUNTRY_CAPACITY) {
+        *err = U_ILLEGAL_ARGUMENT_ERROR;
+        return;
```

Comment:
optional, but like in Google code for readability (and as in the current code) --
```suggestion
    if (langLength >= ULOC_LANG_CAPACITY ||
            scriptLength >= ULOC_SCRIPT_CAPACITY ||
            regionLength >= ULOC_COUNTRY_CAPACITY) {
        *err = U_ILLEGAL_ARGUMENT_ERROR;
        return;
```